packages/atomic/src/components/atomic-search-box/atomic-search-box.tsx

```diff
@@ -476,28 +527,61 @@ export class AtomicSearchBox {
     );
   }
 
-  private renderSuggestions() {
-    const showSuggestions = this.hasSuggestions && this.isExpanded;
+  private async updateSuggestedQuery(q: string) {
+    await Promise.allSettled(
+      this.suggestions.map((suggestion) =>
+        promiseTimeout(
+          suggestion.onSuggestedQueryChange
+            ? suggestion.onSuggestedQueryChange(q)
+            : Promise.resolve(),
+          this.suggestionTimeout
+        )
+      )
+    );
+    forceUpdate(this);
+  }
 
+  private renderSuggestions() {
+    if (!this.hasSuggestions) {
+      return null;
+    }
+    const leftElements = this.getSuggestionElements(this.leftSuggestions);
+    const rightElements = this.getSuggestionElements(this.rightSuggestions);
     return (
-      <ul
-        id={this.popupId}
-        role="listbox"
-        part="suggestions"
-        aria-label={this.bindings.i18n.t('query-suggestion-list')}
-        ref={(el) => (this.listRef = el!)}
-        class={`w-full z-10 absolute left-0 top-full rounded-md bg-background border border-neutral ${
-          showSuggestions ? '' : 'hidden'
+      <div
+        class={`flex w-full z-10 absolute left-0 top-full rounded-md bg-background border border-neutral ${
```

Comment:
I would make sure that every element which has styling also has a part

Reply:
good shout